erpnext/support/doctype/service_level_agreement/service_level_agreement.py

```diff
@@ -101,78 +105,169 @@ def get_service_level_agreement_priority(self, priority):
 			"resolution_time": priority.resolution_time
 		})
 
+	def before_insert(self):
+		# no need to set up SLA fields for Issue dt as they are standard fields in Issue
+		if self.document_type == "Issue":
+			return
+
+		service_level_agreement_fields = get_service_level_agreement_fields()
+		meta = frappe.get_meta(self.document_type, cached=False)
+
+		if meta.custom:
+			self.create_docfields(meta, service_level_agreement_fields)
+		else:
+			self.create_custom_fields(meta, service_level_agreement_fields)
+
+	def on_trash(self):
```

Comment:
Can the custom fields be deleted on trash?

Reply:
Unsure. IMO, deleting fields is a bit risky. If the users have some custom app or have created custom fields by themselves, they wouldn't want them to be deleted.